src/utils/filterOrder.ts

```diff
@@ -0,0 +1,37 @@
+import { Order } from "@cowprotocol/contracts";
+import { BigNumber, ethers } from "ethers";
+
+/**
+ * Process an order to determine if it is valid
+ * @param order The GPv2.Order data struct to validate
+ * @throws Error if the order is invalid
+ */
+export function validateOrder(order: Order) {
+  // amounts must be non-zero
+  if (BigNumber.from(order.sellAmount).isZero()) {
+    throw new Error("Order has zero sell amount");
+  }
+
+  if (BigNumber.from(order.buyAmount).isZero()) {
+    throw new Error("Order has zero buy amount");
+  }
+
+  // token addresses must not be the ZeroAddress
+  if (order.sellToken === ethers.constants.AddressZero) {
+    throw new Error("Order has zero sell token address");
+  }
+
+  if (order.buyToken === ethers.constants.AddressZero) {
+    throw new Error("Order has zero buy token address");
+  }
+
+  // tokens must not be the same
+  if (order.sellToken === order.buyToken) {
+    throw new Error("Order has identical sell and buy token addresses");
+  }
+
+  // Check to make sure that the order has at least 120s of validity
+  if (Math.floor(Date.now() / 1000) + 60 > Number(order.validTo)) {
```

Comment:
Should we include this logic? There will be a duplication of backend logic (not related to the protocol), so we can easily change the API logic and forget to change it here in the future. For example, maybe in Gnosis Chain, they decide to shorten 

nit: Also, this relies on the clock of this service, possibly rejecting an order that the API would accept.

Reply:
okok